command/aarelayer/service/aa_state.go

```diff
@@ -13,58 +15,163 @@ type AATxState interface {
 	Add(*AATransaction) (*AAStateTransaction, error)
 	// Get retrieves the metadata for the AA transaction with the specified ID from the state
 	Get(string) (*AAStateTransaction, error)
-	// Update modifies the metadata for the AA transaction with the specified ID in the state
-	Update(string, func(tx *AAStateTransaction)) error
+	// Get all pending transactions
+	GetAllPending() ([]*AAStateTransaction, error)
+	// Get all pending transactions
+	GetAllQueued() ([]*AAStateTransaction, error)
+	// Update modifies the metadata for the AA transaction
+	Update(stateTx *AAStateTransaction) error
 }
 
+var (
+	pendingBucket  = []byte("pending")
+	queuedBucket   = []byte("queued")
+	finishedBucket = []byte("finished")
+)
+
 var _ AATxState = (*aaTxState)(nil)
 
 type aaTxState struct {
-	// TODO: will be replaced with boldDB in next PR/task
-	mutex sync.RWMutex
-	items map[string]*AAStateTransaction
+	db *bolt.DB
 }
 
-func NewAATxState() (*aaTxState, error) {
-	return &aaTxState{
-		items: make(map[string]*AAStateTransaction),
-	}, nil
+func NewAATxState(dbFilePath string) (*aaTxState, error) {
+	state := &aaTxState{}
+
+	if err := state.init(dbFilePath); err != nil {
+		return nil, err
+	}
+
+	return state, nil
 }
 
 func (s *aaTxState) Add(tx *AATransaction) (*AAStateTransaction, error) {
-	s.mutex.Lock()
-	s.mutex.Unlock()
-
-	// TODO: bolDB implementation will be in next PR/task
-	id := uuid.NewString()
 	ntx := &AAStateTransaction{
-		ID:     id,
+		ID:     uuid.NewString(),
```

Comment:
We could use the hash of the transaction as key perhaps? Just a thought. I am ok with this as well since we are not using this anywhere but to store transactions by some id. 
If we are to later find the `aaTransaction` by the transaction hash, then having hash as a key could be useful. But that is a big if.

Reply:
those transactions are changed during the lifetime (at least status, but also error, gasUsed, etc...). Id should be always the same